ngrinder-controller/src/main/java/org/ngrinder/infra/config/WebMvcConfig.java

```diff
@@ -0,0 +1,154 @@
+		super.addArgumentResolvers(argumentResolvers);
+	}
+
+	@Bean
+	public PageableHandlerMethodArgumentResolver pageableHandlerMethodArgumentResolver() {
+		PageableHandlerMethodArgumentResolver pageableHandlerMethodArgumentResolver = new PageableHandlerMethodArgumentResolver();
+		pageableHandlerMethodArgumentResolver.setPrefix("page.");
+		return pageableHandlerMethodArgumentResolver;
+	}
+
+	@Bean
+	public RemainedPathMethodArgumentResolver remainedPathMethodArgumentResolver() {
+		return new RemainedPathMethodArgumentResolver();
+	}
+
+	@Bean
+	public UserHandlerMethodArgumentResolver userHandlerMethodArgumentResolver() {
+		return new UserHandlerMethodArgumentResolver();
+	}
+
+	@Override
+	public void configureHandlerExceptionResolvers(List<HandlerExceptionResolver> exceptionResolvers) {
+		exceptionResolvers.add(apiExceptionHandlerResolver());
+		exceptionResolvers.add(exceptionHandlerExceptionResolver());
+	}
+
+	@Bean
+	public ApiExceptionHandlerResolver apiExceptionHandlerResolver() {
+		ApiExceptionHandlerResolver apiExceptionHandlerResolver = new ApiExceptionHandlerResolver();
+		apiExceptionHandlerResolver.setOrder(-1);
+		return apiExceptionHandlerResolver;
+	}
+
+	@Bean
+	public ExceptionHandlerExceptionResolver exceptionHandlerExceptionResolver() {
+		ExceptionHandlerExceptionResolver exceptionHandlerExceptionResolver = new ExceptionHandlerExceptionResolver();
+		exceptionHandlerExceptionResolver.setOrder(1);
+		return exceptionHandlerExceptionResolver;
+	}
+
+	@Override
+	public void addInterceptors(InterceptorRegistry registry) {
+		registry.addInterceptor(localeChangeInterceptor);
+	}
+
+	@Bean
+	public MultipartResolver multipartResolver() {
+		CommonsMultipartResolver commonsMultipartResolver = new CommonsMultipartResolver();
+		commonsMultipartResolver.setMaxUploadSize(50000000);
+		commonsMultipartResolver.setDefaultEncoding("utf-8");
+		return commonsMultipartResolver;
+	}
+
+	@Bean
+	public ViewResolver freemarkerResolver() {
+		FreeMarkerViewResolver resolver = new FreeMarkerViewResolver();
+		resolver.setOrder(1);
+		resolver.setViewClass(org.springframework.web.servlet.view.freemarker.FreeMarkerView.class);
+		resolver.setPrefix("");
+		resolver.setSuffix(".ftl");
+		resolver.setRequestContextAttribute("req");
+		Properties properties = new Properties();
+		properties.put("nGrinderVersion" , ngrinderVersion);
+		resolver.setExposeContextBeansAsAttributes(false);
+		resolver.setAttributes(properties);
+		resolver.setContentType("text/html; charset=UTF-8");
+		resolver.setExposeSpringMacroHelpers(true);
+		resolver.setCache(false);
+		return resolver;
+	}
+
+	@Bean
+	@Autowired
+	public FreeMarkerConfigurer freemarkerConfig(ServletContext servletContext) {
+		FreeMarkerConfigurer configurer = new NGrinderFreeMarkerConfigure();
+		configurer.setTemplateLoaderPath("classpath:/templates/ftl/");
+		configurer.setPreferFileSystemAccess(false);
+		configurer.setDefaultEncoding("UTF-8");
```

Comment:
I think this can be set by spring application.properties without creating bean explicitly.